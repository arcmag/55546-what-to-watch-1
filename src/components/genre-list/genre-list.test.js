import * as React from 'react';
import * as renderer from 'react-test-renderer';

import GenreList from './genre-list';

const mock = {
  genres: [`fantasy`, `comedy`],
  genre: `fantasy`,
};

describe(`Test GenreList`, () => {
  it(`renderer`, () => {
    const tree = renderer.create(<GenreList
      setActiveGenre={jest.fn()}
      getFilmsByGenre={jest.fn()}
      selectedGenre={mock.genre}
      genres={mock.genres}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

