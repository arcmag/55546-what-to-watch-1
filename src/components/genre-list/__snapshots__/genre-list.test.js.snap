// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GenreList renderer 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      fantasy
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      comedy
    </a>
  </li>
</ul>
`;
